### PR TITLE
I added the validations and specs

### PR DESCRIPTION
**Pull request by EddyShimwa**
#### For the User model:
- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.
#### For the Post model:
- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    my_post = Post.new(title: 'post title', text: 'post text', comments_counter: 6, likes_counter: 10,
+                       author_id: first_user.id)
+    my_post.save
+
+    my_comment = Comment.new(post: my_post, author: first_user, text: 'Hi comments!')
+
+    my_comment.save
+
+    expect(my_comment.post.comments_counter).to eql(6)
+  end
```
Comment (elisha2kyakpo1) [inline spec/models/post_spec.rb]: - [ ] Let's add more tests for a posts model. We need to test for all the validations and all the callback methods as required by the project instructions.
- [ ] The describe block shows that you are testing the comments model, kindly add tests for checking if a post has a title, not exceeding 250, etc. I have added images for your reference.

tests


specs

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @EddyShimwa, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights

✔️ The PR title and description look good and professional.
✔️ No linter errors.
✔️ All the tests available are passing👏

## Required Changes ♻️
_Check the comments under the review._

- [ ] The __Key features__ section in the readme file is not clear with what was done, (I do not understand what you mean by TBD). Kindly add features you added/did in this milestone.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    context 'when likes_counter is negative' do
+      it 'is invalid' do
+        post = Post.new(title: 'Test Post', text: 'Test post text', author: author, comments_counter: 0,
+                        likes_counter: -1)
+        expect(post).to be_invalid
+        expect(post.errors[:likes_counter]).to include('must be greater than or equal to 0')
+      end
+    end
+  end
```
Comment (DammyShittu) [inline spec/models/post_spec.rb]: - [x] Kindly add tests for the `most_recent_comments` method in your Post model.

- [x] As requested by the initial reviewer, kindly add more descriptive key features in your readme doc. "TBD" is not particularly descriptive.

   image

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @EddyShimwa,

## Highlights
✔️ Model tests

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rotimiazeez — APPROVED** [review]
Hi @EddyShimwa 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good Job working on the required changes 👍🏽
- Desscriptive PR and README file 👍🏽

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**